frame/header-mmr/src/lib.rs

```diff
@@ -253,6 +302,38 @@ pub mod migration {
 	}
 
 	pub fn migrate(new_pallet_name: &[u8]) {
-		frame_support::migration::move_pallet(OLD_PALLET_NAME, new_pallet_name);
+		migration::move_pallet(OLD_PALLET_NAME, new_pallet_name);
+	}
+
+	pub fn initialize_pruning_configuration<T>(step: NodeIndex, block_number: NodeIndex)
+	where
+		T: Config,
+	{
+		<PruningConfiguration<T>>::put(MmrNodesPruningConfiguration {
+			step,
+			progress: 0,
+			last_position: mmr::leaf_index_to_mmr_size(block_number),
+		});
+	}
+
+	pub fn initialize_new_mmr_state<T>(
+		module: &[u8],
+		size: NodeIndex,
+		mmr: Vec<T::Hash>,
+		peak_positions: Vec<NodeIndex>,
+	) where
+		T: Config,
+	{
+		migration::remove_storage_prefix(module, b"MMRCounter", &[]);
+
+		<MmrSize<T>>::put(size);
+
+		for position in peak_positions {
+			<Peaks<T>>::insert(position, mmr[position as usize]);
+		}
```

Comment:
mmr is not possible to pass in from outside, peaks can be calculated real time, and peaks required to be initialized before start pruning. 

Suggest to get the mmr nodes from MMRNodeList, and remove mmr and size.